libvast/src/schema.cpp

```diff
@@ -30,6 +30,8 @@
 
 #include <caf/actor_system_config.hpp>
 
+#include <filesystem>
```

Comment:
When adding this, can you also always check whether `vast/path.hpp` and `vast/directory.hpp` includes can be removed?

Reply:
We can't remove `vast/path.hpp` since one of the `load_schema` overloads uses `vast::path` as its function argument still. I tried to minimize the path conversions, but didn't want to tackle the world as there are other callers of `load_schema` and the transitive call def chain can get out-of-scope for what I was trying to achieve in this MR with `filter_dir` specifically.

`vast/directory.hpp` cannot be removed since that's where `filter_dir` function is defined. Once `vast::directory` class has no uses, I'd suggest we create a `filesystem.hpp` perhaps where some functions that are generally useful could live, such as:

```
load_contents -> currently in `path.hpp` but only works on `vast::path`. Add overload for `std::filesystem::path` until `vast::path` is fully gone to ease transition and minimize path conversions.
filter_dir -> currently in `directory.hpp`
recursive_size -> currently in `directory.hpp`
```

Reply:
Sounds like a plan. Thanks for double-checking!

Reply:
No problem. A natural follow up of this PR would be to change `load_schema` to take a `std::filesystem::path` and change its callers everywhere. I'm not sure how deep that call chain is yet, so I may work on continuing down a path for us to remove `vast::directory` first.

Reply:
That works as well. Yep, `directory` doesn't have too many uses and it depends on `vast::path` since it hands out iterators over them, so it needs to go first in the plan of removing `vast::directory` and `vast::path`.